paddle/fluid/framework/operator.cc

```diff
@@ -973,6 +973,10 @@ Scope* OperatorWithKernel::PrepareData(
     const Scope& scope, const OpKernelType& expected_kernel_key,
     std::vector<std::string>* transfered_inplace_vars,
     RuntimeContext* ctx) const {
+  if (type_ == "is_empty") {
```

Comment:
why need this? Is there a better way? abstraction shouldn't depend on a specific operator implementation.

Reply:
Because the current `is_empty` only implements the CPU kernel. But there is no need to access the input tensor's elements, but just need to get the `dims`.

https://github.com/PaddlePaddle/Paddle/blob/ce093d2d5c9a3fd1dd546cc3c979f660588fdd1e/paddle/fluid/operators/is_empty_op.h#L31-L32

So if the input tensor is on GPU, in fact there is no need to transfer it to CPU. To avoid this and if we don't want to add this `if` statement here, we need to rewrite the `is_empty` op.

Reply:
I think we should rewrite is_empty instead of adding if here.